app/src/main/java/com/example/tkey_android/FirstFragment.java

```diff
@@ -158,6 +164,83 @@ public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
                 throw new RuntimeException(e);
             }
         });
+
+        binding.requestNewShare.setOnClickListener(view1 -> {
+            requireActivity().runOnUiThread(() -> {
+                String userAgent = new WebView(getContext()).getSettings().getUserAgentString();
+                try {
+                    SharetransferModule.requestNewShare(activity.appKey, userAgent, activity.appKey.getShareIndexes().get(0), (result) -> {
+                        if(result instanceof Result.Error) {
+                            renderError(((Result.Error<String>) result).exception);
+                        } else if(result instanceof  Result.Success) {
+                            String requestId = ((Result.Success<String>) result).data;
+                            requireActivity().runOnUiThread(() -> {
+                                binding.resultView.setText("Request Id: " + requestId);
+                                activity.sharedpreferences.edit().putString(REQUEST_ID_ALIAS, requestId).apply();
```

Comment:
It appears that you're doing a share transfer in the same instance of the threshold key right?

It needs to occur between separate instances of threshold keys which share a common service provider and metadata server.

```suggestion
                            String enc_key = ((Result.Success<String>) result).data;
                            requireActivity().runOnUiThread(() -> {
                                binding.resultView.setText("Request Id: " + enc_key);
```